syntax = "proto3";

import "google/protobuf/empty.proto";

service RtcSignaling {
  rpc Subscribe(stream RtcSignalingResponse) returns (stream RtcSignalingRequest) {}
}

message RtcSignalingRequest {
  oneof type {
    google.protobuf.Empty create_offer = 1; 
    string answer_offer = 2; 
  }
}

message RtcSignalingResponse {
  oneof type {
    string create_offer = 1; 
    google.protobuf.Empty answer_offer = 2; 
    google.protobuf.Empty noop = 3;
  }
}

